ee/tabby-webserver/src/hub/mod.rs

```diff
@@ -156,9 +159,78 @@ impl Hub for Arc<HubImpl> {
                     .map(|r| RepositoryConfig::new_named(r.name, r.git_url))
                     .collect()
             });
-        result.unwrap_or_else(|e| {
+        repositories.extend(result.unwrap_or_else(|e| {
             warn!("Failed to fetch repositories: {e}");
             vec![]
-        })
+        }));
+
+        let provider_service = self.ctx.github_repository_provider();
```

Comment:
extract a new method `list_github_provided_repositories`, and invoke it here and fill them into repositories.